### ISSUE TITLE
ULI command in Drush 9.0-dev-dev fails ugly when used with previous syntax

### ISSUE THREAD

**Issue by GBirch** [drush-ops/drush#2866]
Drupal version 8.3.4
PHP 5.6 (cli and apache2)
Ubuntu 16.04
Drush  9.0-dev-dev

**Issuing the command using the previous syntax (as still recommended by drushcommands.com, arghh)**:  
`drush uli XXX`

**Gives me**:  
`http://DOMAIN/user/reset/1/[number]/[token]/login?destination=XXX`

**Following this link gives me the following (user id and domain redacted)**:
```
The website encountered an unexpected error. Please try again later.

InvalidArgumentException: The URI 'base:xxx' is invalid. You must use a valid URI scheme. Use base: for a path, e.g., to a Drupal file that needs the base path. Do not use this for internal paths controlled by Drupal. in Drupal\Core\Utility\UnroutedUrlAssembler->assemble() (line 64 of core/lib/Drupal/Core/Utility/UnroutedUrlAssembler.php).
Drupal\Core\EventSubscriber\RedirectResponseSubscriber->getDestinationAsAbsoluteUrl('XXX', 'http://[DOMAIN]') (Line: 59)
Drupal\Core\EventSubscriber\RedirectResponseSubscriber->checkRedirectUrl(Object, 'kernel.response', Object) (Line: 111)
Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher->dispatch('kernel.response', Object) (Line: 184)
Symfony\Component\HttpKernel\HttpKernel->filterResponse(Object, Object, 1) (Line: 166)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 64)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 67)
Drupal\simple_oauth\HttpMiddleware\BasicAuthSwap->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 50)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 656)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

Suggestion:  either (a) filter the $path argument for validity before adding it to $link, or (b) if $path is an integer and --name is not specified, default to the preexisting syntax (i.e. treat as a uid).

**Comment by weitzman** [comment]
Sorry - I have been asking drushcommands.com to update their index.

I'd accept a PR that did some validation and gave nicer guidance.

**Comment by GBirch** [comment]
Yeah, not like I didn't see that response coming.  Path validation seems near impossible (and I think the error above reflects a core bug - you shouldn't be able to craft a URL that makes Drupal crash like that).  But I think I can make the old syntax work (i.e. drush uli xxx) without changing the expected behavior of the new syntax.

https://github.com/drush-ops/drush/pull/2868

Cheers

**Comment by weitzman** [comment]
Closed in favor of #2868

**Comment by greggles** [comment]
Came here when searching for the error and this helped me understand what to do differently but not completely. 

For anyone else who types `drush uli 16` expecting to login as uid 16 and gets this error the likely answer in drush 9+ is to type `drush uli --uid 16`.
